From 66df864892b6ce00feac9ed4598ceddf87e0f974 Mon Sep 17 00:00:00 2001
From: Chaitanya Kulkarni <kch@nvidia.com>
Date: Wed, 19 Oct 2022 22:40:06 -0700
Subject: [PATCH 1/3] nvme-core: use helper to remove overly long lines

In function __nvme_check_ready() there are overly long lines > 80 char.
Even though it is now allowed to have more lines > 80 char per line it
creates inconsistency and confusion when someone try to add new code.

Add a meaningful function to remove overly long lines.

Signed-off-by: Chaitanya Kulkarni <kch@nvidia.com>
---
 drivers/nvme/host/core.c | 26 ++++++++++++++++++--------
 1 file changed, 18 insertions(+), 8 deletions(-)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index dc4220600585..9bda2122a7fe 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -701,6 +701,21 @@ blk_status_t nvme_fail_nonready_command(struct nvme_ctrl *ctrl,
 }
 EXPORT_SYMBOL_GPL(nvme_fail_nonready_command);
 
+static bool nvme_ctrl_allow_ready_cmd(struct nvme_ctrl *c, struct request *rq)
+{
+	if (!(blk_rq_is_passthrough(rq) && nvme_is_fabrics(req->cmd)))
+		return false;
+
+	switch (req->cmd->fabrics.fctype) {
+	case nvme_fabrics_type_connect:
+	case nvme_fabrics_type_auth_send:
+	case nvme_fabrics_type_auth_receive:
+		return true;
+	default:
+		return false;
+	}
+}
+
 bool __nvme_check_ready(struct nvme_ctrl *ctrl, struct request *rq,
 		bool queue_live)
 {
@@ -725,16 +740,11 @@ bool __nvme_check_ready(struct nvme_ctrl *ctrl, struct request *rq,
 		 */
 		switch (ctrl->state) {
 		case NVME_CTRL_CONNECTING:
-			if (blk_rq_is_passthrough(rq) && nvme_is_fabrics(req->cmd) &&
-			    (req->cmd->fabrics.fctype == nvme_fabrics_type_connect ||
-			     req->cmd->fabrics.fctype == nvme_fabrics_type_auth_send ||
-			     req->cmd->fabrics.fctype == nvme_fabrics_type_auth_receive))
-				return true;
-			break;
-		default:
-			break;
+			return nvme_ctrl_allow_ready_cmd(ctrl, rq);
 		case NVME_CTRL_DEAD:
 			return false;
+		default:
+			break;
 		}
 	}
 
-- 
2.29.0

